### ISSUE TITLE
Support Remote Inference for Prompt Guard?

### ISSUE THREAD

**Issue by ckavili** [meta-llama/llama-stack#2998]
### 🚀 Describe the new functionality needed

Hi 👋,

I was running `Prompt-Guard-86M` as a remote service (via FastAPI), and I noticed that the current `Prompt Guard` [implementation](https://github.com/meta-llama/llama-stack/blob/main/llama_stack/providers/inline/safety/prompt_guard/prompt_guard.py) is designed to run only as a local model. It seems to expect direct access to logits for scoring, which makes sense for local use, but becomes a blocker for remote setups. Especially compared to `Llama Guard`, which supports remote inference through the API layer.

In my case, I’ve wrapped `Prompt-Guard-86M` behind an OpenAI-compatible endpoint that returns a structured score and label like this:

```json
{
  "choices": [
    {
      "text": "[JAILBREAK] (0.8234)",
      "metadata": {
        "label": "JAILBREAK",
        "score": 0.8234
      }
    }
  ]
}
```

The key information is still there, just not coming from logits. So I was wondering:

* Is there any interest or plan to allow Prompt Guard provider to work with remote models?
* Would a hybrid mode (local or remote, similar to Llama Guard) be something worth exploring?
* Would a simple response adapter or parser that could read from structured remote responses be a viable option like the example above?

Thanks for all the great work 🙏

### 💡 Why is this needed? What if we don't build it?

While Prompt-Guard-86M is quite lightweight, it still needs to be downloaded or baked into the container image(unless I missed something).

This adds an extra step to deployment, either rebuilding the existing images with the model files, or ensuring the environment has the model in the expected path (possibly via initContainer by providing HF token and running `llama download --source huggingface --model-id Prompt-Guard-86M ` or mounting a PVC etc). Supporting remote inference would eliminate this friction and align Prompt Guard’s behavior with Llama Guard provider.
